signer/src/transaction_coordinator.rs

```diff
@@ -1672,76 +1695,295 @@ where
         })
     }
 
-    /// TODO(#742): This function needs to filter deposit requests based on
-    /// time as well. We need to do this because deposit requests are locked
-    /// using OP_CSV, which lock up coins based on block height or
-    /// multiples of 512 seconds measure by the median time past.
+    /// Fetches pending withdrawal requests from storage and filters them based
+    /// on the remaining consensus rules as defined in #741:
+    ///
+    /// 1. [x] The request must not have been swept within the current canonical
+    ///    Bitcoin chain.
+    /// 2. [x] The request must be confirmed in a canonical Stacks block.
+    /// 3. [x] The request must have reached the required number of Bitcoin
+    /// 4. [x] The request must be approved:
+    ///     - [x] By the required number of signers (this is implemented as a
+    ///       pre-filter in the query, any signer),
+    ///     - [x] And by the required number of signers _in the current signer
+    ///       set_.
+    /// 5. [ ] The request has been approved by this signer. **Note:** This rule
+    ///     does not apply within the coordinator module, where decisions are
+    ///     made collectively based on consensus rules rather than an individual
+    ///     signer's approval. However, the coordinator's signer module still
+    ///     processes the request according to these same rules.
+    /// 6. [ ] The assessed fees will be within the constraints of the request's
+    ///    specified maximum fee (this is handled during packaging).
+    /// 7. [x] The request must not have expired (handled in the query).
+    /// 8. [x] The request amount must be above the dust limit.
+    /// 9. [x] The request must be within the current sBTC caps.
     #[tracing::instrument(skip_all)]
-    pub async fn get_pending_requests(
-        &self,
-        bitcoin_chain_tip: &model::BitcoinBlockRef,
-        stacks_chain_tip: &model::StacksBlockHash,
-        aggregate_key: &PublicKey,
-        signer_public_keys: &BTreeSet<PublicKey>,
-    ) -> Result<Option<utxo::SbtcRequests>, Error> {
-        tracing::info!("preparing pending requests for processing");
-
-        let storage = self.context.get_storage();
-        let config = self.context.config();
-
-        // Get the current sBTC limits (caps).
-        let sbtc_limits = self.context.state().get_current_limits();
+    pub async fn get_eligible_pending_withdrawal_requests<DB>(
+        storage: &DB,
+        expiry_window: u64,
+        soft_expiry_window: u64,
```

Comment:
nit: replace `soft_expiry_window` with `expiry_buffer`?

Reply:
I actually think, when reading the method "from an outside perspective", that `soft_expiry_window` gives more immediate context as to its intent, and that maybe `expiry_window` should be renamed to `hard_expiry_window` or something. Thoughts?

Reply:
I mainly flagged this because of the constant we passed in, now that we removed `WITHDRAWAL_BLOCKS_SOFT_EXPIRY` it could be fine leaving as is.
I find it still a bit confusing since _window_ refer to something slightly different in the two cases so it could be a bit confusing at first -- ie, `soft_expiry_window` is a window relative to `expiry_window` and not the request height -- and in the const (and related discussion) we call it a buffer.
But anyway, no strong opinions about it, do as you prefer.

Reply:
Oh I see what you mean, okay yeah I agree with you, what's passed here isn't a window.

I think it would be clearer if the calculation for the "soft expiry height" were done in `get_pending_requests()` instead and that both were passed in as absolute block heights, instead of mixing the two concepts. But to avoid changing that right now (and tests) I'll just rename this to buffer 👍

Reply:
Tweaked in [5c4edf1](https://github.com/stacks-network/sbtc/pull/1399/commits/5c4edf14c0e5780d2a8dda6d86839e89fce235cd)